BrainPortal/lib/boutiques_input_copier.rb

```diff
@@ -0,0 +1,223 @@
+    parent_input_ids = descriptor.custom_module_info('BoutiquesInputCopier')
+
+    inputs      = descriptor["inputs"]
+
+    # In parent_input_ids, select all non hidden inputs
+    parent_input_ids = parent_input_ids.select { |_, config| !config["hide"] }
+    copied_info = fill_copied_info(parent_input_ids, descriptor, inputs)
+
+    # Add the new inputs to the descriptor and to the group if needed
+    extracted_idx = copied_info.values.map { |info| info[:input_idx] }.sort.reverse
+
+    extracted_idx.each do |index|
+      # In copied_info, find the one with input_idx == index
+      copied_info.each do |parent_inputid, info|
+        next if info[:hide]
+        next if info[:input_idx] != index
+
+        # Insert the new input in the descriptor
+        new_input = info[:new_input]
+        inputs.insert(index, new_input)
+
+        # Insert the new input in the group
+        next if !info[:member_info]
+        group        = descriptor.group_by_id(info[:member_info][:id])
+        members      = group["members"]
+        new_input_id = new_input["id"]
+        next if members.include?(new_input_id)
+        members.insert(info[:member_info][:idx], new_input_id)
+      end
+    end
+
+    descriptor
+  end
+
+  def descriptor_for_form #:nodoc:
+    descriptor_for_before_form()
+  end
+
+  def descriptor_for_after_form #:nodoc:
+      descriptor_for_before_form()
+  end
+
+  ############################################
+  # Bourreau (Cluster) Side Modifications
+  ############################################
+
+  # For input in +BoutiquesInputCopier+ section,
+  def setup #:nodoc:
+    # Log revision information
+    basename = Revision_info.basename
+    commit   = Revision_info.short_commit
+
+    descriptor = self.descriptor_for_setup
+    # First call the main setup
+    return false if !super
+    return true  if descriptor.custom_module_info('BoutiquesInputCopier').blank?
+
+    boutiques_input_copier = descriptor.custom_module_info('BoutiquesInputCopier')
+    invoke_params          = self.invoke_params
+
+    boutiques_input_copier.each do |parent_inputid, config|
+      to_be_copied = self.invoke_params.delete("#{parent_inputid}_copy")
+
+      # Special case if default-value is true and the field is hidden
+      if to_be_copied.blank?
+        to_be_copied = true if config["default-value"] && config["hide"]
+      end
+
+      # Skip if the copy option is not selected
+      next if to_be_copied.blank? || to_be_copied == "0"
+
+      userfile_id = invoke_params[parent_inputid]
+      next if userfile_id.blank?
+      userfile    = Userfile.find(userfile_id)
+      copy_to                                        = userfile.name + "_" + self.run_id.to_s
+      self.invoke_params[parent_inputid]             = copy_to
+      descriptor.input_by_id(parent_inputid)["type"] = "String"
+
+      rsync_cmd = "sleep 10; rsync -a -L --no-g --chmod=u=rwX,g=rX,Dg+s,o=r --delete #{userfile.name.bash_escape}/* #{copy_to.bash_escape}"
```

Comment:
`sleep 10` ?!?!

Reply:
Why use a pattern with `/*` ? This will not copy any files that starts with `.` which might still be an miportant part of the files being copied